zulipterminal/core.py

```diff
@@ -187,7 +187,7 @@ def _narrow_to(self, button: Any, anchor: Optional[int],
         if already_narrowed:
             return
 
-        self.model.found_newest = False
+        self.model.found_newest[repr(narrow)] = False
```

Comment:
I understand this is just an update to existing code, but it would be great if we could treat `found_newest` as an attribute private to the Model; if we could do that refactor first, this commit would also be smaller and cleaner.

Directly setting data in another class like this is something I'm always open to removing, as it typically leads to simplification and removes the need to check between classes when updating code - and this is a small repo!